File: Courzelo_Back/src/main/java/tn/esprit/courzelo/entities/ProjectEntities/Project.java
package tn.esprit.courzelo.entities.ProjectEntities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import tn.esprit.courzelo.entities.UserCorzelo.Speciality;
import tn.esprit.courzelo.entities.UserCorzelo.UserCourzelo;
import java.util.Set;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "Project")
public class Project {
    @Id
    private String id;
    @Indexed
    private String name;
    @Indexed
    private String description;
    @Indexed
    private Difficulty difficulty;

    private Set<Speciality> specialities;
    @Indexed
    private Date datedebut;
    @Indexed
    private Date deadline;
    @Indexed
    private int number;
    @Indexed
    private boolean hasGroupProject;

    @DBRef
    private List<Tasks> tasks;
    @DBRef
    private List<UserCourzelo> users;
    @DBRef
    private List<GroupProject> groupProjects;


}

File: Courzelo_Front/Courzelo/src/app/Models/ProjectEntities/Project.ts
import { Speciality } from "../UserCorzelo/UserCourzelo";
import { Tasks } from "./Tasks";

export class Project {
    id!: number;
    name!: string;
    description!: string;
    difficulty!: Difficulty;
    datedebut!: Date;
    deadline!: Date;
    number!: number;
    hasGroupProject!: boolean;
    tasks!:Array<Tasks>;


    // Define the specialities array within the class
    specialities: Speciality[] = [
        Speciality.BD,
        Speciality.Angular,
        Speciality.Spring,
        Speciality.DotNet,
        Speciality.Reseau,
        Speciality.IA,
        Speciality.Mobile,
        Speciality.Web,
        Speciality.Cloud,
        Speciality.DevOps,
        Speciality.Security,
        Speciality.Design,
        Speciality.Management,
        Speciality.Marketing,
        Speciality.Finance
    ];
}

export enum Difficulty {
    Hard, Easy, Medium
}
